heartbeat/autodiscover/builder/hints/heartbeats.go

```diff
@@ -0,0 +1,211 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package hints
```

Comment:
It feels odd to name this file `heartbeats.go`, That implies that it has some plural relationship to `heartbeat`. Maybe `hints.go`?

Reply:
i just followed the convention for logs and metrics hints builder in filebeat and metricbeat.

Reply:
Ah, I see. Looking at metric beat it named its file `metrics`. Given that I think the right name here is `monitors.go`. It seems the convention is to name the file the same as the noun of the thing it is generating, which in heartbeat's case is a monitor.